.github/workflows/build.yml

```diff
@@ -143,13 +88,16 @@ jobs:
     - name: Record dependencies
       id: record-deps
       run: |
+        # The tests call out to msys2 commands. We generally do not want to mix toolchains, so
+        # we are very deliberate about only adding msys64 to the path where absolutely necessary.
+        ${{ (runner.os == 'Windows' && '$env:PATH=("C:\msys64\mingw64\bin;{0}" -f $env:PATH)') || '' }}
```

Comment:
We add this to almost every section, is it necessary?  Can we alternatively have commands that do not depend on this, then modify `PATH` and then commands which require it?

Reply:
Fixed!